### ISSUE TITLE
Training is not stable for custom dataset using DeepLabV3+ model

### ISSUE THREAD

**Issue by sayanmandal** [GeorgeSeif/Semantic-Segmentation-Suite#92]
Please fill out this issue template before submitting. **Issues which do not fill out this template, or are already answered in the FAQs will simply be closed.**

**Please go to Stack Overflow for help and support.** Also check past issues as many are repeats. Also check out the Frequently Asked Questions (FAQs) below in case your question has already been answered in an issue!

Issues should be one of the following:

1. Feature requests
2. Bug reports

------------------------

## Please specify the following information when submitting an issue:
- **What are your command line arguments?:**
 python train.py --dataset Carla --model DeepLabV3_plus
- **Have you written any custom code?**:
No
- **What have you done to try and solve this issue?:**
I have changed the dataset to suit our need as described in the README. (I have also included class_dict.csv) according to our need.
- **TensorFlow version?**:
1.4.0

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug or a feature request.

I am trying to run the DeepLabV3+ model for the Carla simulator dataset that we have collected. The class_dict.csv has been populated from the Carla official repository.[( https://github.com/carla-simulator/carla/blob/master/PythonClient/carla/image_converter.py)] But the training was not found to be stable. I am attaching the loss here after epoch 1:

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

[2018-09-09 02:09:14] Epoch = 1 Count = 20 Current_Loss = 0.4596 Time = 5.37
[2018-09-09 02:09:20] Epoch = 1 Count = 40 Current_Loss = 0.4578 Time = 5.70
[2018-09-09 02:09:26] Epoch = 1 Count = 60 Current_Loss = 0.4171 Time = 5.64
[2018-09-09 02:09:31] Epoch = 1 Count = 80 Current_Loss = 0.4324 Time = 5.24
[2018-09-09 02:09:36] Epoch = 1 Count = 100 Current_Loss = 1.5930 Time = 5.18
[2018-09-09 02:09:42] Epoch = 1 Count = 120 Current_Loss = 4.2199 Time = 5.44
[2018-09-09 02:09:47] Epoch = 1 Count = 140 Current_Loss = 32.1963 Time = 5.57
[2018-09-09 02:09:53] Epoch = 1 Count = 160 Current_Loss = 582.5015 Time = 5.53
[2018-09-09 02:09:58] Epoch = 1 Count = 180 Current_Loss = 119388.2969 Time = 5.38
[2018-09-09 02:10:03] Epoch = 1 Count = 200 Current_Loss = 1257567.2500 Time = 5.18
[2018-09-09 02:10:09] Epoch = 1 Count = 220 Current_Loss = 3359160.5000 Time = 5.52
[2018-09-09 02:10:14] Epoch = 1 Count = 240 Current_Loss = 9596012.0000 Time = 5.21

Let me know if you need to know anything about the problem.
### FAQs

- **Question:** I got an `InvalidArgumentError` saying that `Dimensions of inputs should match` **Answer:** See issue #17

- **Question:** Can you upload pre-trained weights for these networks? **Answer:** See issue #57

- **Question:** Do I need a GPU to train these models? **Answer:** Technically no, but I'd highly recommend it. I was able to train the models pretty well in about a day using a 1080Ti GPU. Training on CPU would take much longer than that.

- **Question:** Will you be adding the FCN or U-Net models? **Answer:** No I won't be adding those simply because they're a few years old and state-of-the-art has moved past that.

- **Question:** I got an invalid argument error when using the InceptionV4 model. Am I doing something wrong? **Answer:** No you're not! Due to the design of the InceptiveV4 model, when you end up upsampling you do some rounding which creates a shape mismatch. _This only happens when you end up having to use the `end_points['pool5']`_. See the code for some of the models if you want to check whether the model will use `end_points['pool5']`.

**Comment by GeorgeSeif** [comment]
Well since the loss is growing, it looks like your learning rate it too high! Try bringing that down a bit.

Closing since this is not a bug or feature request but feel free to comment again if you have anymore trouble :)
